enclave-runtime/src/extrinsic_daemon.rs

```
@@ -0,0 +1,230 @@
+		let latest_parentchain_header = v.latest_finalized_header(v.num_relays())?;
+		let genesis_hash = v.genesis_hash(v.num_relays())?;
+		Ok((latest_parentchain_header, genesis_hash))
+	})?;
+	let authority = Ed25519Seal::unseal_from_static_file()?;
+	let node_metadata_repository = GLOBAL_NODE_METADATA_REPOSITORY_COMPONENT.get()?;
+
+	let extrinsics_factory = ExtrinsicsFactory::new(
+		genesis_hash,
+		authority,
+		GLOBAL_NONCE_CACHE.clone(),
+		node_metadata_repository,
+	);
+
+	let author_api = GLOBAL_TOP_POOL_AUTHOR_COMPONENT.get()?;
+
+	let state_handler = GLOBAL_STATE_HANDLER_COMPONENT.get()?;
+	let state_observer = GLOBAL_STATE_OBSERVER_COMPONENT.get()?;
+	// For debug purposes, list shards. no problem to panic if fails.
+	let shards = state_handler.list_shards().unwrap();
+	let default_shard_identifier: ShardIdentifier = if let Some(shard) = shards.get(0) {
+		Ok(sp_core::H256::from_slice(shard.as_bytes()))
+	} else {
+		Err(Error::Stf("Could not retrieve shard".to_string()))
+	}?;
+
+	let stf_state: StfState =
+		state_handler.load(&default_shard_identifier).map_err(Error::StfStateHandler)?;
+	let stf_state = Arc::new(stf_state);
+
+	let shielding_key_repository = GLOBAL_SHIELDING_KEY_REPOSITORY_COMPONENT.get()?;
+	let shielding_key = Rsa3072Seal::unseal_from_static_file().unwrap();
+
+	let ocall_api = GLOBAL_OCALL_API_COMPONENT.get()?;
+
+	let stf_enclave_signer =
+		Arc::new(EnclaveStfEnclaveSigner::new(state_observer, ocall_api, shielding_key_repository));
+
+	let request_context = RequestContext::new(
+		default_shard_identifier,
+		stf_state,
+		shielding_key,
+		stf_enclave_signer,
+		author_api,
+	);
+	loop {
+		let request_type = receiver.recv().map_err(|e| Error::Other(e.into()))?;
+
+		match request_type {
+			RequestType::Web2IdentityVerification(ref request) => {
+				if let Err(e) = web2_identity_verification(&request_context, request.clone()) {
+					error!("Could not retrieve data from https server due to: {:?}", e);
```

I'm a bit concerned about the error handling, we should propagate the error to parentchain so that F/E is notified, but it's worth a separate ticket.